src/downloader.h

```diff
@@ -0,0 +1,20 @@
+#ifndef AKTUALIZR_LITE_DOWNLOADER_H_
+#define AKTUALIZR_LITE_DOWNLOADER_H_
+
+#include "aktualizr-lite/api.h"
+
+class Downloader {
+ public:
+  virtual DownloadResult Download(const TufTarget& target) = 0;
```

Comment:
I'm guessing there are some pros/cons for using TufTarget here instead of Uptane::Target here. I get this worry we may go through a phase of code where we have different call-sites having to convert between the two. Can you explain the rationale and if you have an idea for a boundary between the two?

Reply:
The only reason why I use TufTarget because we've has a plan to move towards using the api for all clients, specifically, we planned to move aklite's main.cc to the api usage?.
If it's so, then eventually it will just one component that does the conversion - LiteClient, interface of which will have just TufTarget, the conversion will be encapsulated in LiteClient implementation.

Reply:
Yep, it makes sense. The reason I did so is that I thought that Downloader ifc would become a part of the API some day  so a user can define/inject its own  downloader :). But, I think, I am kind of projecting the experience I got at HERE, their API is more flexible and allows their clients many different use-cases (e.g. custom download, download and installation are decoupled).
Anyway, I think you are right, it doesn't make sense to do it now, so the conversion will be encapsulated in liteclient.*.

Reply:
Actually I have second thoughts on this matter, I think the current implementation is a good trade-off as for an initial step towards minimizing and encapsulating dependency on aktualizr in general and not using the package manager interface at all in particular.
I can imagine many implementation options so intermediate steps towards the goal, so let's talk about it verbally to avoid confusion and assumptions about each other thoughts.

Reply:
Sure, eventually the conversion will be encapsulated in implementation of the interface(s) that abstract LiteClient from aktualizr.